components/custom_board/tas5805m/tas5805m.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * ESPRESSIF MIT License
+ *
+ * Copyright (c) 2020 <ESPRESSIF SYSTEMS (SHANGHAI) CO., LTD>
+ *
+ * Permission is hereby granted for use on all ESPRESSIF SYSTEMS products, in
+ * which case, it is free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the
+ * Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ */
+
+#include "tas5805m.h"
+#include "esp_log.h"
+#include "i2c_bus.h"
+#include "tas5805m_reg_cfg.h"
+
+static const char *TAG = "TAS5805M";
+
+/* Default I2C config */
+
+static i2c_config_t i2c_cfg = {
+    .mode = I2C_MODE_MASTER,
+    .sda_pullup_en = GPIO_PULLUP_ENABLE,
+    .scl_pullup_en = GPIO_PULLUP_ENABLE,
+    .master.clk_speed = I2C_MASTER_FREQ_HZ,
+};
+
+/*
+ * Operate fuction of PA
+ */
+audio_hal_func_t AUDIO_CODEC_TAS5805M_DEFAULT_HANDLE = {
+    .audio_codec_initialize = tas5805m_init,
+    .audio_codec_deinitialize = tas5805m_deinit,
+    .audio_codec_ctrl = tas5805m_ctrl,
+    .audio_codec_config_iface = tas5805m_config_iface,
+    .audio_codec_set_mute = tas5805m_set_mute,
+    .audio_codec_set_volume = tas5805m_set_volume,
+    .audio_codec_get_volume = tas5805m_get_volume,
+    .audio_hal_lock = NULL,
+    .handle = NULL,
+};
+
+/* Init the I2C Driver */
+
+void i2c_master_init()
+{
+  int i2c_master_port = I2C_MASTER_NUM;
+
+  get_i2c_pins(I2C_NUM_0, &i2c_cfg);
+
+  esp_err_t res = i2c_param_config(i2c_master_port, &i2c_cfg);
+  printf("Driver param setup : %d\n", res);
+  res = i2c_driver_install(i2c_master_port, i2c_cfg.mode,
+                           I2C_MASTER_RX_BUF_DISABLE, I2C_MASTER_TX_BUF_DISABLE,
+                           0);
+  printf("Driver installed   : %d\n", res);
```

Comment:
Please use ESP LOG always

Reply:
Is done in the next commit.

Reply:
Did you change this already?

Reply:
I think here I messed up while debugging. I will change it to ESP_LOGI again. I was happy to validate that the idf5 i2s driver works with the inversion so I don't keep my eyes on that ;-)